FRONT-21 Add MapScreen with Google Maps

JacobRoe opened the pull request:
Adds a MapScreen to the the frontend that uses the Google Maps API behind the scene (via @react-google-maps/api). 

The API Key needs to be provided via the environment variable REACT_APP_GOOGLE_MAPS_API_KEY. Because the backend, as far as I know, doesn't provide the data for the locations yet, dummy data is created to make some markers visible.

### Furthermore, there are some tasks left to do:
- [ ] 1. Add the correct icons for the markers
- [ ] 2. Align the position of the info popups according to the marker position
- [ ] 3. Update MapScreen when data from backend is available
- [ ] 4. Minor refactorings (e.g. extract Markers, remove unused imports)

Task 2 will be resolved as soon as the [Pull Request](https://github.com/JustFly1984/react-google-maps-api/pull/879) at @react-google-maps/api gets merged. Otherwise, I could add the correct positioning myself, if it's necessary because of deadline reasons.

### Appendix
In this image the hands represent the requests and disinfectants the offers.
![image](https://user-images.githubusercontent.com/13980359/83031919-7a130880-a035-11ea-98dd-0683c761d4f8.png)

JacobRoe committed:
Add MapScreen with Google Maps

hauptmedia commented:
@matusmala @dominikhorn93 How can we get location data from backend for the maps view? I think we also miss an opt-in function for the user to enable the map function for their offer/demand?

JacobRoe committed:
Merge branch 'develop' into feat/map-integration

JacobRoe committed:
Update MapScreen to use real backend data

Offers or needs from the same place should be represented as one marker, otherwise the user is not able to select all ones. This change contains a temporary implementation of the idea. It should possibly be handled by the backend.

JacobRoe commented:
Updated the MapScreen. See commit message.

Together with the PR from https://github.com/remedyMatch/rm-backend/pull/28, the map is now able to show offers and needs from the backend as markers. Offers and needs from the exact same location are united into one marker.

marcoebbinghaus commented:
> @matusmala @dominikhorn93 How can we get location data from backend for the maps view? I think we also miss an opt-in function for the user to enable the map function for their offer/demand?

Could you explain your question in more detail? What is needed from the backend for the maps view? What do you mean with "location data"? We save geocoding information in Institution#standorte (each standort consisting of name, strasse, hausnummer, plz, ort, land, latitude, longitude). This is where we save those location information.
